### PR TITLE
chore(Portal): Convert to RFC

### PR DESCRIPTION
**Pull request by assuncaocharles**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Include a change request file using `$ yarn change`

# BREAKING CHANGES

Converting `Portal` from class component to functional. 

Related to #12237

#### Focus areas to test

(optional)

### CODE REVIEW

File: packages/fluentui/react-northstar/src/components/Portal/Portal.tsx [hunk 1]
```diff
@@ -78,118 +86,127 @@ export interface PortalState {
...
-    const { children, content, trapFocus } = this.props;
-    const { open } = this.state;
-
+const Portal: React.FC<WithAsProp<PortalProps>> & FluentComponentStaticProps<PortalProps> = props => {
+  const context: ProviderContextPrepared = React.useContext(ThemeContext);
+  const { setStart, setEnd } = useTelemetry(Portal.displayName, context.telemetry);
+  setStart();
+  const { children, content, trapFocus, trigger, triggerAccessibility } = props;
+  let portalNode: HTMLElement;
+  let triggerNode: HTMLElement;
```
Comment (layershifter) [inline packages/fluentui/react-northstar/src/components/Portal/Portal.tsx:95]:
```suggestion
  const portalNode = React.useRef<HTMLElement>();
  const triggerNode = React.useRef<HTMLElement>();
```

File: packages/fluentui/react-northstar/src/components/Portal/Portal.tsx [hunk 2]
```diff
@@ -78,118 +86,127 @@ export interface PortalState {
...
+  });
+
+  const renderPortal = (): JSX.Element | undefined => {
     const contentToRender = childrenExist(children) ? children : content;
     const focusTrapZoneProps = (_.keys(trapFocus).length && trapFocus) || {};
-
     return (
       open && (
-        <Ref innerRef={this.handlePortalRef}>
+        <Ref innerRef={handlePortalRef}>
```
Comment (layershifter) [inline packages/fluentui/react-northstar/src/components/Portal/Portal.tsx:108]:
```suggestion
        <Ref innerRef={portalRef}>
```

File: packages/fluentui/react-northstar/src/components/Portal/Portal.tsx [hunk 3]
```diff
@@ -78,118 +86,127 @@ export interface PortalState {
...
-  renderPortal(): JSX.Element | undefined {
-    const { children, content, trapFocus } = this.props;
-    const { open } = this.state;
-
+const Portal: React.FC<WithAsProp<PortalProps>> & FluentComponentStaticProps<PortalProps> = props => {
+  const context: ProviderContextPrepared = React.useContext(ThemeContext);
+  const { setStart, setEnd } = useTelemetry(Portal.displayName, context.telemetry);
+  setStart();
+  const { children, content, trapFocus, trigger, triggerAccessibility } = props;
+  const portalNode = React.useRef<HTMLElement>();
```
Comment (layershifter) [inline packages/fluentui/react-northstar/src/components/Portal/Portal.tsx:94]:
```suggestion
  const portalRef = React.useRef<HTMLElement>();
```

File: packages/fluentui/react-northstar/src/components/Portal/Portal.tsx [hunk 4]
```diff
@@ -78,118 +86,127 @@ export interface PortalState {
...
+  const triggerNode = React.useRef<HTMLElement>();
```
Comment (layershifter) [inline packages/fluentui/react-northstar/src/components/Portal/Portal.tsx:95]:
```suggestion
  const triggerRef = React.useRef<HTMLElement>();
```

File: packages/fluentui/react-northstar/src/components/Portal/Portal.tsx [hunk 5]
```diff
@@ -78,118 +86,127 @@ export interface PortalState {
...
+
+  const handleUnmount = () => {
+    _.invoke(props, 'onUnmount', props);
   };
 
-  handlePortalRef = (portalNode: HTMLElement) => {
-    this.portalNode = portalNode;
+  const handlePortalRef = (node: HTMLElement) => {
+    portalNode = node;
   };
```
Comment (layershifter) [inline packages/fluentui/react-northstar/src/components/Portal/Portal.tsx:147]:
```suggestion
```

File: packages/fluentui/react-northstar/src/components/Portal/Portal.tsx [hunk 6]
```diff
@@ -78,118 +86,127 @@ export interface PortalState {
...
 
-  handleTriggerRef = (triggerNode: HTMLElement) => {
-    this.triggerNode = triggerNode;
-    handleRef(this.props.triggerRef, triggerNode);
+  const handleTriggerRef = (node: HTMLElement) => {
+    triggerNode = node;
```
Comment (layershifter) [inline packages/fluentui/react-northstar/src/components/Portal/Portal.tsx:149]:
```suggestion
    triggerRef.current = node;
```

File: packages/fluentui/react-northstar/src/components/Portal/Portal.tsx [hunk 7]
```diff
@@ -78,118 +86,127 @@ export interface PortalState {
...
+    handleRef(props.triggerRef, triggerNode);
```
Comment (layershifter) [inline packages/fluentui/react-northstar/src/components/Portal/Portal.tsx:150]:
```suggestion
    handleRef(props.triggerRef, node);
```

File: packages/fluentui/react-northstar/src/components/Portal/Portal.tsx [hunk 8]
```diff
@@ -78,118 +86,127 @@ export interface PortalState {
...
   };
 
-  handleTriggerClick = (e: React.MouseEvent, ...unhandledProps) => {
-    const { trigger } = this.props;
+  const handleTriggerClick = (e: React.MouseEvent, ...unhandledProps) => {
+    const { trigger } = props;
```
Comment (layershifter) [inline packages/fluentui/react-northstar/src/components/Portal/Portal.tsx:151]:
```suggestion
```

File: packages/fluentui/CHANGELOG.md [hunk 9]
```diff
@@ -28,6 +28,7 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 - Restricted prop sets in the `Form` component which are passed to styles functions, @assuncaocharles ([#13180](https://github.com/microsoft/fluentui/pull/13180))
 - Restricted prop sets in the `FormField` component which are passed to styles functions, @assuncaocharles ([#13181](https://github.com/microsoft/fluentui/pull/13181))
 - Restricted prop sets in the `DropdownSelectedItem` component which are passed to styles functions, @assuncaocharles ([#13179](https://github.com/microsoft/fluentui/pull/13179))
+- Restricted prop sets in the `Portal` component which are passed to styles functions, @assuncaocharles ([#13186](https://github.com/microsoft/fluentui/pull/13186))
```
Comment (mnajdova) [inline packages/fluentui/CHANGELOG.md:31]: This is a false breaking change

**Pull request merged**